src/aleph/chains/common.py

```diff
@@ -365,37 +366,42 @@ async def get_chaindata_messages(chaindata, context, seen_ids: Optional[List]=No
         try:
-            content, size = await get_json(chaindata["content"], timeout=10)
+            content = await get_json(chaindata["content"], timeout=10)
+        except AlephStorageException:
+            # Let the caller handle unavailable/invalid content
+            raise
         except Exception:
-            LOGGER.exception(
-                "Can't get content of offchain object %r" % chaindata["content"]
-            )
-            return None
-        if content is None:
-            return None
-
-        messages = await get_chaindata_messages(content, context)
-        if messages is not None and messages != -1:
-            LOGGER.info("Got bulk data with %d items" % len(messages))
-            if app["config"].ipfs.enabled.value:
-                # wait for 4 seconds to try to pin that
-                try:
-                    LOGGER.info(f"chaindatax {chaindata}")
-                    await PermanentPin.register(multihash=chaindata["content"],
-                                                reason={
-                                                    "source": "chaindata",
-                                                    "protocol": chaindata["protocol"],
-                                                    "version": chaindata["version"],
-                                                })
-                    await asyncio.wait_for(pin_hash(chaindata["content"]), timeout=4.0)
-                except asyncio.TimeoutError:
-                    LOGGER.warning(f"Can't pin hash {chaindata['content']}")
-        else:
+            error_msg = f"Can't get content of offchain object {chaindata['context']!r}"
+            LOGGER.exception("%s", error_msg)
+            raise ContentCurrentlyUnavailable(error_msg)
+
+        try:
+            messages = await get_chaindata_messages(content.value, context)
+        except AlephStorageException:
             LOGGER.debug("Got no message")
+            raise
+
+        LOGGER.info("Got bulk data with %d items" % len(messages))
+        if app["config"].ipfs.enabled.value:
+            # wait for 4 seconds to try to pin that
+            try:
+                LOGGER.info(f"chaindatax {chaindata}")
+                await PermanentPin.register(
+                    multihash=chaindata["content"],
+                    reason={
+                        "source": "chaindata",
+                        "protocol": chaindata["protocol"],
+                        "version": chaindata["version"],
+                    },
+                )
+                await asyncio.wait_for(pin_hash(chaindata["content"]), timeout=4.0)
+            except asyncio.TimeoutError:
+                LOGGER.warning(f"Can't pin hash {chaindata['content']}")
         return messages
     else:
-        LOGGER.info("Got unknown protocol/version object in tx %r" % context)
-        return -1
+        error_msg = f"Got unknown protocol/version object in tx {context!r}"
+        LOGGER.info("%s", error_msg)
```

Comment:
```suggestion
        LOGGER.info(""Got unknown protocol/version object in tx '%s'", f"{error_msg!r}")
```
```

Reply:
`f"{error_msg!r}"` -> `repr(error_msg)`? Or  is there a reason to use an f-string I'm not aware of?

Reply:
That's better indeed. I naively copied from your string.